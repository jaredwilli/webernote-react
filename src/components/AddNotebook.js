import React from 'react';

import CloseBtn from './ui/CloseBtn';

const AddNotebook = (props) => {
    return (
        <span className="add-notebook">
            <input type="text" name="notebook"
                className="new-notebook"
                placeholder="Notebook name"
                onBlur={(e) => props.addNotebook(e)}
                onKeyDown={(e) => props.keyPress(e)} />

            {(props.notebooks && props.notebooks.length > 0) &&
                <CloseBtn onClick={(e) => props.toggleAddState(e)} />
            }
        </span>
    );
}

export default AddNotebook;
