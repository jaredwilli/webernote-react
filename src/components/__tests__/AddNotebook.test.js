import React from 'react';
import renderer from 'react-test-renderer';

import AddNotebook from '../AddNotebook';

describe('AddNotebook', () => {
    it('should match the snapshot', () => {
        const notebooks = [];
        const funcs = {
            addNotebook: function() {},
            keyPress: function() {},
            toggleAddState: function() {}
        };

        const component = renderer.create(<AddNotebook
            notebooks={notebooks}
            addNotebook={funcs.addNotebook}
            keyPress={funcs.keyPress}
            toggleAddState={funcs.toggleAddState} />).toJSON();

        expect(component).toMatchSnapshot();
    });
});
