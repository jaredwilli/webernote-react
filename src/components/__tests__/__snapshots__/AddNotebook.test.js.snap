// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNotebook should match the snapshot 1`] = `
<span
  className="add-notebook"
>
  <input
    className="new-notebook"
    name="notebook"
    onBlur={[Function]}
    onKeyDown={[Function]}
    placeholder="Notebook name"
    type="text"
  />
</span>
`;
